### ISSUE TITLE
thread 'rustc' panicked at 'no entry found for key', src/libcore/option.rs:1187:5

### ISSUE THREAD

**Issue by darcyjcain** [rust-lang/rust#66742]
Rust version is 1.41.0.20191124.

stack backtrace:
   0:        0x801129137 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h211b928ed0f2e497
   1:        0x80115fed1 - core::fmt::write::hb7790f3f63b223a9
   2:        0x801120922 - std::io::Write::write_fmt::h27bf71b7b9673f5f
   3:        0x80112a746 - std::panicking::default_hook::{{closure}}::h33a90a230154e8dc
   4:        0x80112a3e1 - std::panicking::default_hook::h9ed982ced4849844
   5:        0x801c5c522 - rustc_driver::report_ice::hd49d184a267f5b9d
   6:        0x80112af23 - std::panicking::rust_panic_with_hook::h2d729cf939a7ab12
   7:        0x80112a9e3 - std::panicking::continue_panic_fmt::h66bc5771b5ba805c
   8:        0x80112a8c9 - rust_begin_unwind
   9:        0x801166e5d - core::panicking::panic_fmt::h8753411d440ce26f
  10:        0x80114e61f - core::option::expect_failed::h74914dfc3b691ee3
  11:        0x80353129d - rustc::hir::map::def_collector::DefCollector::collect_field::hb6d2b9aa475e90a4
  12:        0x8020da71d - syntax_expand::expand::AstFragment::visit_with::h92691c51459da66f
  13:        0x80207637d - rustc_resolve::macros::<impl syntax_expand::base::Resolver for rustc_resolve::Resolver>::visit_ast_fragment_with_placeholders::h7072677902e63e9e
  14:        0x802f4325d - syntax_expand::expand::MacroExpander::collect_invocations::hf06c197df32f5f0c
  15:        0x802f40bfd - syntax_expand::expand::MacroExpander::fully_expand_fragment::h651baef1f65e5c4f
  16:        0x802f3b726 - syntax_expand::expand::MacroExpander::expand_crate::h991da3332087b767
  17:        0x801db6ee6 - rustc_interface::passes::configure_and_expand_inner::{{closure}}::hbc8ae5eece938790
  18:        0x801db5224 - rustc_interface::passes::configure_and_expand_inner::hb686b9ce6b9a89d3
  19:        0x801cefc22 - rustc_interface::passes::configure_and_expand::{{closure}}::h6478fe0ad9c0fa68
  20:        0x801cc7caa - rustc_data_structures::box_region::PinnedGenerator<I,A,R>::new::h9cf3bfc50966f55f
  21:        0x801db3503 - rustc_interface::passes::configure_and_expand::ha799fd680b6d9ae7
  22:        0x801d821c9 - rustc_interface::queries::Query<T>::compute::h2a920182257a80a6
  23:        0x801d308ec - rustc_interface::queries::<impl rustc_interface::interface::Compiler>::expansion::h115c8c2759ff513c
  24:        0x801c70e26 - rustc_interface::interface::run_compiler_in_existing_thread_pool::h96a49b62f2eeafdc
  25:        0x801c1d837 - std::thread::local::LocalKey<T>::with::h580e3de029daabe9
  26:        0x801c1649a - scoped_tls::ScopedKey<T>::set::h59fc89b873641542
  27:        0x801c3d367 - syntax::with_globals::h851721207f8ca16f
  28:        0x801c17931 - std::sys_common::backtrace::__rust_begin_short_backtrace::hc0ab8933bbb75761
  29:        0x80112ecff - __rust_maybe_catch_panic
  30:        0x801c32a25 - core::ops::function::FnOnce::call_once{{vtable.shim}}::hb15a1639b3794d1d
  31:        0x80112008f - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h68e2c036f5dbb999
  32:        0x801121021 - std::sys_common::thread::start_thread::hdc635b1cb81895aa
  33:        0x801106229 - std::sys::unix::thread::Thread::new::thread_start::hcadad9c6737c7dfe
  34:        0x801187736 - <unknown>

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.41.0-nightly running on x86_64-unknown-freebsd

note: compiler flags: -C opt-level=2 -C panic=abort -C codegen-units=1 -C linker=/usr/obj/usr/ports/www/firefox/work/firefox-70.0.1/build/cargo-linker -C opt-level=2 -C force-frame-pointers=yes --crate-type lib

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
end of query stack
error: could not compile `style`.

Caused by:
  process didn't exit successfully: `CARGO_PKG_VERSION_PATCH=1 CARGO_PKG_VERSION_MAJOR=0 CARGO=/usr/local/bin/cargo CARGO_PKG_NAME=style CARGO_PKG_DESCRIPTION= CARGO_PKG_VERSION_MINOR=0 CARGO_PKG_VERSION=0.0.1 CARGO_PKG_HOMEPAGE= LD_LIBRARY_PATH='/usr/obj/usr/ports/www/firefox/work/.build/release/deps:/usr/local/lib' CARGO_PKG_VERSION_PRE= CARGO_PKG_REPOSITORY= CARGO_MANIFEST_DIR=/usr/obj/usr/ports/www/firefox/work/firefox-70.0.1/servo/components/style OUT_DIR=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/build/style-e6a3063ff63d1f99/out CARGO_PKG_AUTHORS='The Servo Project Developers' /usr/local/bin/rustc --crate-name style servo/components/style/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C opt-level=2 -C panic=abort -C codegen-units=1 --cfg 'feature="bindgen"' --cfg 'feature="gecko"' --cfg 'feature="nsstring"' --cfg 'feature="regex"' --cfg 'feature="toml"' -C metadata=f0aad9c4077a198e -C extra-filename=-f0aad9c4077a198e --out-dir /usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps --target x86_64-unknown-freebsd -C linker=/usr/obj/usr/ports/www/firefox/work/firefox-70.0.1/build/cargo-linker -L dependency=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps -L dependency=/usr/obj/usr/ports/www/firefox/work/.build/release/deps --extern app_units=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libapp_units-d8bffcb82975fb08.rmeta --extern arrayvec=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libarrayvec-a8b5dbd6793451a1.rmeta --extern atomic_refcell=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libatomic_refcell-d9d425ee3653475b.rmeta --extern bitflags=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libbitflags-ab3ead12c9134009.rmeta --extern byteorder=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libbyteorder-4990ba4f67190314.rmeta --extern cssparser=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libcssparser-8feb184ac168f51a.rmeta --extern derive_more=/usr/obj/usr/ports/www/firefox/work/.build/release/deps/libderive_more-a224960bb5ba8fe7.so --extern euclid=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libeuclid-e25ac98082a22fbf.rmeta --extern fallible=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libfallible-0446974b0a55b640.rmeta --extern fxhash=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libfxhash-20c1e5c96d8eb421.rmeta --extern hashglobe=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libhashglobe-4ce63bd1ecae9dce.rmeta --extern indexmap=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libindexmap-119fae5f3de8879f.rmeta --extern itertools=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libitertools-5c9c6ff88736148f.rmeta --extern itoa=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libitoa-3e873034535babc2.rmeta --extern lazy_static=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/liblazy_static-bf11fa9f2e3c2923.rmeta --extern log=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/liblog-575063ab4931e5d2.rmeta --extern malloc_size_of=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libmalloc_size_of-d325b083f1fcbb4b.rmeta --extern malloc_size_of_derive=/usr/obj/usr/ports/www/firefox/work/.build/release/deps/libmalloc_size_of_derive-618184ffae90b87f.so --extern matches=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libmatches-8b527c52aac53ee4.rmeta --extern debug_unreachable=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libdebug_unreachable-abc1399f48ec58a1.rmeta --extern nsstring=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libnsstring-91789eea315ec810.rmeta --extern num_derive=/usr/obj/usr/ports/www/firefox/work/.build/release/deps/libnum_derive-5a30caa78c8caf7c.so --extern num_integer=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libnum_integer-84804ee03fbcfb0c.rmeta --extern num_traits=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libnum_traits-7d2d5d395f209c65.rmeta --extern num_cpus=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libnum_cpus-94812777192666a4.rmeta --extern ordered_float=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libordered_float-a3cd06928a273cd1.rmeta --extern owning_ref=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libowning_ref-5944dd4e2862896d.rmeta --extern parking_lot=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libparking_lot-3f9453696cda9848.rmeta --extern precomputed_hash=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libprecomputed_hash-79a12c0ed4bff28a.rmeta --extern rayon=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/librayon-28451143065c806d.rmeta --extern selectors=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libselectors-08327d81842bd2af.rmeta --extern servo_arc=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libservo_arc-feab3b27733d7983.rmeta --extern smallbitvec=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libsmallbitvec-2ad9b130602d3302.rmeta --extern smallvec=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libsmallvec-d3353d9d9cb62594.rmeta --extern static_prefs=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libstatic_prefs-aaa79ea4cf9e02d2.rmeta --extern style_derive=/usr/obj/usr/ports/www/firefox/work/.build/release/deps/libstyle_derive-c999698aeb10f400.so --extern style_traits=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libstyle_traits-5c9cf1ae410a828c.rmeta --extern thin_slice=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libthin_slice-dfea6fdcb04c8018.rmeta --extern time=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libtime-b2f5854a1998db2e.rmeta --extern to_shmem=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libto_shmem-087447a488774424.rmeta --extern to_shmem_derive=/usr/obj/usr/ports/www/firefox/work/.build/release/deps/libto_shmem_derive-37f0daac2b4a925b.so --extern uluru=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libuluru-b3e41ca925fa0962.rmeta --extern unicode_bidi=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libunicode_bidi-0c2f3c0e5e1587f2.rmeta --extern unicode_segmentation=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libunicode_segmentation-b2cee7797651c3ac.rmeta --extern void=/usr/obj/usr/ports/www/firefox/work/.build/x86_64-unknown-freebsd/release/deps/libvoid-3f44c92c68ddb156.rmeta -C opt-level=2 -C force-frame-pointers=yes --cap-lints warn` (exit code: 101)

**Comment by jonas-schievink** [comment]
Please provide the source code needed to reproduce this

**Comment by darcyjcain** [comment]
On 2019-11-25 9:39 a.m., Jonas Schievink wrote:
> Please provide the source code needed to reproduce this

I have no source code.  That was the error when I tried to build Firefox
using the FreeBSD ports.  The message asked for a bug report so that's
what I did.

-- 
D'Arcy J.M. Cain <darcy@druid.net>         |  Democracy is three wolves
http://www.druid.net/darcy/                |  and a sheep voting on
+1 416 788 2246     (DoD#0082)    (eNTP)   |  what's for dinner.
IM: darcy@Vex.Net, VoIP: sip:darcy@druid.net
